src/Microsoft.OData.Core/Json/JsonLightInstanceAnnotationWriter.cs

```diff
@@ -394,7 +398,7 @@ await this.WriteInstanceAnnotationNameAsync(propertyName, annotationName)
                 await this.valueSerializer.WriteResourceValueAsync(resourceValue,
                     expectedType,
                     treatLikeOpenProperty,
-                    this.valueSerializer.CreateDuplicatePropertyNameChecker()).ConfigureAwait(false);
+                    this.valueSerializer.JsonLightOutputContext.DuplicatePropertyNameChecker).ConfigureAwait(false);

```

Comment:
At a different level, it could be a "concurrency" problem?

I mean when we write a payload with multiple "levels", for example, Top Resource,  Child resource, GrandChild resource...

The PropertyNameChecker is reused in each "level"?  If yes, it's a big problem if Top Resource has the same property name with GrandChild resource?